server/src/main/java/org/jboss/as/server/Main.java

```diff
@@ -91,14 +91,16 @@ public static void main(String[] args) {
             }
 
             Module.registerURLStreamHandlerFactoryModule(Module.getBootModuleLoader().loadModule(ModuleIdentifier.create("org.jboss.vfs")));
-            ServerEnvironment serverEnvironment = determineEnvironment(args, WildFlySecurityManager.getSystemPropertiesPrivileged(),
+            ServerEnvironmentWrapper serverEnvironmentWrapper = determineEnvironment(args, WildFlySecurityManager.getSystemPropertiesPrivileged(),
                     WildFlySecurityManager.getSystemEnvironmentPrivileged(), ServerEnvironment.LaunchType.STANDALONE,
                     Module.getStartTime());
-            if (serverEnvironment == null) {
-                abort(null);
+            if (serverEnvironmentWrapper.getServerEnvironment() == null) {
+                if (serverEnvironmentWrapper.getServerEnvironmentStatus() == ServerEnvironmentWrapper.ServerEnvironmentStatus.ERROR) {
+                    abort(null);
+                }
```

Comment:
We need an 'else' clause here where we still end up calling SystemExiter.logAndExit(), just with an ExitLogger that's a no-op.